### PR TITLE
Add hook in the processIndexPHP method for a very early plugin entryp…

### PR DESCRIPTION
**Pull request by amichelsdatabay**
…oint

### CODE REVIEW

**Comment by mjansenDatabay** [comment]
@amichelsdatabay  Please check your indent style. Please use tabs, **not** spaces.

File: Services/Init/classes/class.ilStartUpGUI.php [hunk 1]
```diff
@@ -1807,7 +1807,24 @@ protected function showTermsOfService()
 	*/
 	protected function processIndexPHP()
 	{
-		global $ilIliasIniFile, $ilAuth, $ilSetting;
+		global $ilIliasIniFile, $ilAuth, $ilSetting, $DIC;
+
+        //
+        $ilPluginAdmin = null;
+        if (isset($DIC["ilPluginAdmin"]))
```
Comment (chfsx) [inline Services/Init/classes/class.ilStartUpGUI.php:1814]: You could use `$DIC->offsetExists("ilPluginAdmin");` instead

File: Services/Init/classes/class.ilStartUpGUI.php [hunk 2]
```diff
@@ -1807,7 +1807,24 @@ protected function showTermsOfService()
...
+        {
+            $ilPluginAdmin = $DIC["ilPluginAdmin"];
+        }
+
+        // include the user interface hook
+        if (is_object($ilPluginAdmin)) {
```
Comment (chfsx) [inline Services/Init/classes/class.ilStartUpGUI.php:1820]: How about a check for `$ilPluginAdmin instanceof ilPluginAdmin`

**Review by chfsx — COMMENTED** [review]
Just my two cents

File: Services/Init/classes/class.ilStartUpGUI.php [hunk 3]
```diff
@@ -1807,7 +1807,24 @@ protected function showTermsOfService()
 	*/
 	protected function processIndexPHP()
 	{
-		global $ilIliasIniFile, $ilAuth, $ilSetting;
+		global $ilIliasIniFile, $ilAuth, $ilSetting, $DIC;
+
+        //
```
Comment (mjansenDatabay) [inline Services/Init/classes/class.ilStartUpGUI.php:1812]: @amichelsdatabay Please remove this line.

File: Services/Init/classes/class.ilStartUpGUI.php [hunk 4]
```diff
@@ -1807,7 +1807,24 @@ protected function showTermsOfService()
 	*/
 	protected function processIndexPHP()
 	{
-		global $ilIliasIniFile, $ilAuth, $ilSetting;
+		global $ilIliasIniFile, $ilAuth, $ilSetting, $DIC;
```
Comment (mjansenDatabay) [inline Services/Init/classes/class.ilStartUpGUI.php:1810]: @amichelsdatabay You could use the `$DIC` to also fetch `$ilIliasIniFile, $ilAuth, $ilSetting`.

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]

**Comment by mjansenDatabay** [comment]
@amichelsdatabay What is the purpose of this? Do you need this to manipulate the generated HTML? Oder do you just need some kind of hook/event?

**Review by alex40724 — CHANGES_REQUESTED** [review]
Hi, thank you for this pr. However this more looks like a request for the event Hook plug-in slot. And this point no HTML is generated. Could you explain the purpose of this request in more detail? Thanks.

**Comment by amichelsdatabay** [comment]
I took the UI hook as seen in the ilUtil::redirect method because this hook is used to replace (the visible part) the routing of ILIAS.
But i see that multiple slots will fit here. Please decide yourself (or with the technical board) which is the best slot. I will change it to your choice.

File: Services/Init/classes/class.ilStartUpGUI.php [hunk 5]
```diff
@@ -1807,7 +1807,24 @@ protected function showTermsOfService()
...
+
+        // include the user interface hook
+        if (is_object($ilPluginAdmin)) {
+            $pl_names = $ilPluginAdmin->getActivePluginsForSlot(IL_COMP_SERVICE, "UIComponent", "uihk");
+            foreach ($pl_names as $pl) {
+                $ui_plugin = ilPluginAdmin::getPluginObject(IL_COMP_SERVICE, "UIComponent", "uihk", $pl);
+                $gui_class = $ui_plugin->getUIClassInstance();
+                $resp = $gui_class->getHTML("Services/Init", "processIndexPHP", array(  ));
+            }
+        }
```
Comment (alex40724) [inline Services/Init/classes/class.ilStartUpGUI.php:1827]: Pleas check your indentation style

**Comment by alex40724** [comment]
Again, could you please explain the purpose of this request in more detail? If you cannot do this, I have to simply close this. What is the use case? Why does it use the getHTML method? Why is $resp not being processed?

I think the existing code in the "redirect" procedure is a misuse of the interface, since it does not deal with HTML but uses getHTML and modifyHTML. It would be better to extend the interface of the slot with additional methods, imo. I guess this would also be appropriate for your case. But since the purpose is unclear to me, I cannot make any suggestions.

**Comment by mjansenDatabay** [comment]
@amichelsdatabay Can you please give some feedback here?

**Comment by PurHur** [comment]
Matter has been solved using the hook within the ilUtil redirect method. Sadly this is far from being a good practice. Kindly acknowledge that these discussion are anything but a productive approach in developing a good piece of art (software ;-) ).
Will not participate any further in this discussion, just treat this matter as closed.

Edit: Wrong Account i know but i dont have my other credentials.

**Comment by chfsx** [comment]
@alex40724 this seems to be outdated since @amichelsdatabay is no longer interested in changing this.

**Pull request closed without merging**